File: src/test/java/uk/gov/justice/digital/hmpps/oauth2server/security/OracleSha1PasswordEncoderTest.java
package uk.gov.justice.digital.hmpps.oauth2server.security;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class OracleSha1PasswordEncoderTest {
    @Test
    void matches_testMatchesOraclePassword() {
        final var encodedPassword = "S:39BA463D55E5C8936A6798CC37B1347BA8BEC37B6407397EB769BC356F0C";
        assertThat(new OracleSha1PasswordEncoder().matches("somepass1", encodedPassword)).isTrue();
    }

    @Test
    void matches_invalidPasswordLength() {
        final var encodedPassword = "somepasswordthatshouldbeencoded";
        assertThat(new OracleSha1PasswordEncoder().matches("somepass1", encodedPassword)).isFalse();
    }

    @Test
    void matches_nullInput() {
        assertThat(new OracleSha1PasswordEncoder().matches("somepass1", null)).isFalse();
    }
}

File: src/test/java/uk/gov/justice/digital/hmpps/oauth2server/security/OracleSha1PasswordEncoderTest.kt
package uk.gov.justice.digital.hmpps.oauth2server.security

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class OracleSha1PasswordEncoderTest {
  @Test
  fun matches_testMatchesOraclePassword() {
    val encodedPassword = "S:39BA463D55E5C8936A6798CC37B1347BA8BEC37B6407397EB769BC356F0C"
    assertThat(OracleSha1PasswordEncoder().matches("somepass1", encodedPassword)).isTrue()
  }

  @Test
  fun matches_invalidPasswordLength() {
    val encodedPassword = "somepasswordthatshouldbeencoded"
    assertThat(OracleSha1PasswordEncoder().matches("somepass1", encodedPassword)).isFalse()
  }

  @Test
  fun matches_nullInput() {
    assertThat(OracleSha1PasswordEncoder().matches("somepass1", null)).isFalse()
  }
}
